pages/education.py

```diff
@@ -432,9 +431,7 @@ def render_public_private_figure(private_data, public_data, country):
         merged["private_percentage"] = merged["real_expenditure_private"] / (
             merged["real_expenditure_private"] + merged["real_expenditure_public"]
         )
-        merged["public_percentage"] = merged["real_expenditure_public"] / (
-            merged["real_expenditure_private"] + merged["real_expenditure_public"]
-        )
+        merged["public_percentage"] = 1 - merged["private_percentage"]
 
```

Comment:
I think I missed this in my previous PR, but I think we need to add an explicity empty check (for country like Burkina faso without private education data).

```
if merged.empty:
            raise Exception("No data available for this period")
```

Reply:
@yukinko-iwasaki I fixed it with some more improvements. If you recall, using error handling for flow control is a code smell as it hides real exception cases and goes against the "fail fast" rule. I removed the try-except and I think the rest of the file should be refactored too. Probably a backlog item for a separate PR.

Reply:
hmm.. I thought we could go with EAFP style since it was in python. But I am also OK with if/else type of approach.

Reply:
EAFP is used for exception handling while what we have here isn't an exception/error case; it's flow control. The difference is that we are not checking for error, but a specific condition where the dataset is empty.